### ISSUE TITLE
Laura's review feedback

### ISSUE THREAD

**Issue by lauracowen** [OpenLiberty/draft-guide-microprofile-reactive-messaging-rest#19]
I didn't see the previous versions of the intro but I think this makes sense to me now. The following feedback mostly just makes some of the wording/explanations a bit clearer or are edits for style:

## What you'll learn section
(btw, your headings look fine as they are - I've just abbreviated them in this issue to save myself typing)
- "You will write microservices so that" -> "…some microservices…"? or "…the microservices…"? Not sure what microservices this is referring to as the RESTful ones already exist? Should it say "You will modify the two microservices in the application in this guide so that…"?
- "producer events are generated" by what?
- "how we can achieve code execution on demand" -> "how we can run code on demand" - do we really see this? What in the guide demonstrates this? I'm not sure this sentence adds anything…
- I think you should split this long paragraph into two shorter paragraphs before "The current inventory…." sentence. Then insert the diagram between them with a lead-in sentence of "The following diagram illustrates the application:" Then have the more detailed description of what's shown in the diagram after the diagram - it makes it easier to visualise what you're referring to.
- If the microservices are called "system" and "inventory" (lower-case), the diagram should use the same capitalisation when labelling them.
- how does a PUT request response query a system property? Isn't it the request that queried the system property and the response contains the response to that query? This wording doesn't make sense (and is hard to parse when reading).

## Adding a REST endpoint section
- don't think we should keep saying "hit an endpoint" (or variants like "hittable"!). I know developer colloquially say that but it's weird. "access an endpoint" is more accurate?
- "by replacing the InventoryResource class." "by replacing the InventoryResource class with an updated version of the class." (otherwise it just seems weird that we're telling them to replace the class - with what?)
- I think be more explicit when it says "Add the RESTful endpoint to the inventory service" eg -> "Add a RESTful endpoint called `/data` to the inventory service…"
- "The changes include…" paragraph reads oddly - the first sentence summarises the changes but then the second sentence randomly picks one of those items to explain. I actually don't think this first sentence is necessary - it's just introducing a bunch of things at once without explaining them. This isn't a bad approach in general writing but I don't think it adds anything here. It'd be clearer to just remove it and start that paragraph with "The updateSystemProperty() method creates the /data endpoint, which…" nicely introduces how the /data endpoint (which you've now mentioned above the action box anyway). Then the following paragraphs address each of the other relevant methods/variables in turn without the reader having to keep track of what was introduced earlier.
- Then move the "propertyNameEmitter" sentence up to the end of the previous paragraph as it's referring to the same property name request as the propertyNameEmitter sentence so it flows nicely.
- Then the second paragraph explains the sendPropertyName() method and the Publisher stuff.
- "creates the /data endpoint that accepts PUT requests " -> "creates the /data endpoint which accepts PUT requests " (which, not that,)
- "The propertyNameEmitter is a RxJava Emitter" -> "The propertyNameEmitter variable is an RxJava Emitter" - don't use the name of something as the noun (it won't translate well and is potentially confusing what you're referring to; also, use "an" when the following sound is like a vowel - say it out loud: I think it's pronounced "Ar-eks-Java", not "Rex-Java"?)
- "that will be responsible" -> "that is responsible"; similarly "which controls what the Publisher will do if" -> "which controls what the Publisher does if"; "which will let the Publisher buffer" -> "which lets the Publisher buffer " - use present tense when describing how the technology works (only use future tense when you're talking about actions within the context of the guide eg you will do x later in this guide, you will learn. Mostly, no future tense except in the What you'll learn section, as a rule of thumb.) Go through the rest of the guide and fix other similar instances. Also "the Publisher used the…which will let…" -> "the Publisher uses the…which lets…"
- consistently lower-case words like "emitter", "publisher", "subscriber", except when referring to an actual class name. There's no need to keep capitalising them - it's distracting to read.
- "utilizing" -> "using" (no need to use fancy words when a simpler one will do).

## Adding an event processor section
- You can simplify the wording here slightly too to make it easier to read: "The changes include creating a method named sendProperty() which receives a system property name from the inventory microservice over the @Incoming("propertyRequest") channel, calculates the requested property on the fly and publishes it back to Kafka over the @Outgoing("propertyResponse") channel. " -> "A new method named sendProperty( receives a system property name from the inventory microservice over the @Incoming("propertyRequest") channel. The method calculates the requested property on the fly and publishes it back to Kafka over the @Outgoing("propertyResponse") channel. "

## Configuring connectors section
- Simplify wording: "The newly created RESTful endpoint requires new channels to be created to move the requested messages between the system and inventory microservices. " -> "The newly created RESTful endpoint requires two new channels which move the requested messages between the system and inventory microservices. "
- Try to avoid saying "Replacing" in the text (it's fine in the action boxes but it reads oddly because what you're doing is updating the microservice code really - it's just that the action in the context of the guides is to replace the contents of a file with new lines that we've also supplied. "Replacing the inventory microservice microprofile-config.properties file adds the two channels, requestSystemProperty and addSystemProperty to handle sending the system property request, and receiving the system property response respectively." -> "The inventory microservice microprofile-config.properties file now has two new channels, requestSystemProperty and addSystemProperty, which handle sending the system property request and receiving the system property response respectively."
- You don't need to say "Replace the system microservice microprofile-config.properties file to" because the action text has already said that. Just describe what the changes are here: "Two new channels, propertyRequest and propertyResponse,now handle receiving the property request and sending the property response respectively."
- "NOTE : If you want the message to be sent to all the System microservices then configure the unique group.id for all of the instances."
  - Don't use "NOTE:". Just have the sentence if it's important to mention (start with "If you want…").
  - However, I'm not sure what this sentence is saying - "all of the instances" of what? (the System microservice? We only have one in this app.) And how would you configure the group.id - one line per system microservice?
  - I don't think this is actually necessary to mention at all though. We only have one here. And this kind of diverts from the path through the guide.
  - If you want to make this a kind of stretch goal from which to learn, after the Testing section, you could maybe add a Next steps section like we have in the draft Social Login guide (http://lgdev1.fyre.ibm.com:4020/guides/social-login.html#next-steps) which starts by giving a little context something like "This application has only one instance of the system microservice." Then suggests what they could do as an exercise without giving all the detailed steps or code eg "You could try creating another instance (on a different port??) and adding another `group.id` line in the xxxxproperties file so that the message from the xxx is received by both instances of the system microservice." Or something? Or just leave it out completely.

## Testing section
- don't use future tense when describing how the tech works
- The "For example :" line is odd. And I think we can simplify this: "….existing properties. For example, run the following curl command:" then have the command to copy/paste. Then "In this example, the PUT request with the os.name system property…."
- In what circumstances would curl be unavailable? Windows OSes? (I have it on my Mac, Linux machines will have it. Does Windows not?)
- If a significant number of users are likely not to have curl (eg all Windows users?), it's sufficient to say "If the curl command is unavailable on your computer, use another client such as <Postman>, which enables you to make requests using a graphical interface." Don't explain how to use Postman. That's not what we're here for and it seems pretty straightforward - we need to assume they undersatnd how to make a PUT request (so remove the rest of the paragraph). Are we okay essentially recommending Postman (as opposed to anything else)? Check with YK.
- Reword slightly: "Since the system service is available, the request to the service is successful" -> "The system service is available so the request to the service is successful…"

**Comment by yeekangc** [comment]
I concur that the latest version is looking good.  Thank you, @MaiHameed @gkwan-ibm.

**Comment by MaiHameed** [comment]
> In what circumstances would curl be unavailable? Windows OSes? (I have it on my Mac, Linux machines will have it. Does Windows not?) 

Windows does not by default, I changed this section to use OS tabs and point to using other methods in the Windows section. Sidenote for @gkwan-ibm We'll need to change this as we used the same Postman explanation in multiple guides

**Comment by MaiHameed** [comment]
Hi @lauracowen thanks again for the feedback,  I opened a PR #20, site  has been updated with the latest changes http://lgdev1.fyre.ibm.com:4020/guides/microprofile-reactive-messaging-rest.html

**Comment by lauracowen** [comment]
Nice! :) Thanks Mai. I've approved the PR on the basis of the fyre draft.
